libcst/_nodes/expression.py

```diff
@@ -1020,6 +1020,23 @@ def _codegen_impl(self, state: CodegenState) -> None:
             self.whitespace_between._codegen(state)
             self.right._codegen(state)
 
+    @property
+    def evaluated_value(self) -> Optional[str]:
+        """
+        Return an :func:`ast.literal_eval` evaluated str of recursively concatenated :py:attr:`left` and :py:attr:`right`
+        if and only if both :py:attr:`left` and :py:attr:`right` are composed by :class:`SimpleString` or :class:`ConcatenatedString`
+        (:class:`FormattedString` cannot be evaluated).
+        """
+        left = self.left
+        right = self.right
+        if isinstance(left, FormattedString) or isinstance(right, FormattedString):
+            return None
+        left_val = left.evaluated_value
+        right_val = right.evaluated_value
+        if right_val is None:
```

Comment:
Why is this check only for right_val?

Reply:
https://libcst.readthedocs.io/en/latest/nodes.html#libcst.ConcatenatedString
`left` can be either SimpleString or FormattedString, we just check it's not an FormattedString, so it must be a SimpleString. SimpleString.evaluated_value is always a str.

The `right` part can be a ConcatenatedString, which can have None as the evaluated_value whenever it contains any FormattedString. So we only apply None check on `right` here.